core/core-impl/src/test/java/ai/timefold/solver/core/config/solver/EnvironmentModeTest.java

```diff
@@ -147,15 +155,23 @@ void corruptedUndoShadowVariableListener(EnvironmentMode environmentMode) {
 
         switch (environmentMode) {
             case FULL_ASSERT:
-            case FAST_ASSERT:
```

Comment:
We need a name for the new environment mode, something more descriptive than `SUPER_FULL_ASSERT`. Or maybe not?

Your original numbers suggest that this is not *that much* slower than `FULL_ASSERT`. If you can first fix the score calculation numbers to give us *correct* data, we can assess those speeds again and if we're talking about a drop of a couple percent, we really don't need a whole new env mode for this.